lib/Versions/ExpireManager.php

```diff
@@ -117,6 +118,16 @@ public function getExpiredVersion(array $versions, int $time, bool $quotaExceede
 		});
 
 		$expired = array_filter($versionsLeft, function (IVersion $version) use ($quotaExceeded) {
+			// Do not expire current version.
+			if ($version instanceof IMetadataVersion && $version->getTimestamp() === $version->getSourceFile()->getMtime()) {
+				return false;
+			}
+
+			// Do not expire versions with a label.
+			if ($version->getMetadataValue('label') !== null && $version->getMetadataValue('label') !== '') {
```

Comment:
It’s a bit weird/sad that there is not a method like isExpirable in versions instead. It seems a bit arbitrary to decide from groupfolders app what the rules are for keeping versions.